client/internal/http/request.go

```diff
@@ -46,6 +48,9 @@ func NewRequest(config cdo.Config, httpClient *http.Client, logger *log.Logger,
 		method: method,
 		url:    url,
 		body:   body,
+
+		Header:      make(http.Header),
```

Comment:
Why don't you make this an array, so you can pass in multiple headers? Why restrict to just one?

Reply:
Actually, it does support multiple headers, the name that golang chose is quite confusing, but `http.Header` is actually a `map[string][]string` (map of string to string array), it has a method call `Add` which is actually an `append` internal to the string array. So it is more similar to [Guava's multimap](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Multimap.html) interface